standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -5239,12 +5236,9 @@ public List<Partition> alterPartitions(String catName, String dbName, String tbl
       tblName = normalizeIdentifier(tblName);
       List<MPartition> mPartitionList;
 
-      try (Query query = pm.newQuery(MPartition.class,
-              "table.tableName == t1 && table.database.name == t2 && t3.contains(partitionName) " +
-                      " && table.database.catalogName == t4")) {
-        query.declareParameters("java.lang.String t1, java.lang.String t2, java.util.Collection t3, "
-                + "java.lang.String t4");
-        mPartitionList = (List<MPartition>) query.executeWithArray(tblName, dbName, partNames, catName);
+      Pair<Query, Map<String, String>> queryWithParams = getPartQueryWithParams(catName, dbName, tblName, partNames);
```

Comment:
I see that now the sql query has become partitionName= p1 || partitionName=p2 instead of partitionName.contains(p1). I'm wondering how this would impact 10k+ partitions or 1M partitions.

Reply:
@VenuReddy2103 Can you please answer this comment if you are able to? This PR was closed because it became stale due to inactivity. I have revived it, so we can re-review and merge it when the tests are clean.